internal/daemon/plugin/vendorplugin.go

```diff
@@ -24,6 +24,11 @@ import (
 //go:embed bindata/*
 var binData embed.FS
 
+var VspImages = []string{
+	"IntelVspPlugin",
```

Comment:
> Yes, that implies to import "ipu.go" module to places that need nothing except knowing the name of the variable.

If you dislike that, add the variable to `vendorplugin.go`. The point is, to use the string literal only at one place, and let the compiler be aware of where the `IntelVspPluginVariable` (or whatever we call it) is used.

maybe even make it an enum or a struct for plugin metadata with multiple fields (like a `prettyname`, `env_var_name`, `factory_method` etc). But at least, use a constant for the string literal. But there are already factory classes, those should expose such metadata (like the name of the environment variable).

It's a good step to add a `VspImages` list. But this should be a list of "useful and typed metadata about plugins".

Reply:
We need to decide first if we want to create this from the daemon or from the controller. If it's from the daemon, you can return a blob or use the Platform api *as* the blob.

Reply:
I think it has to be deployed from the daemon, otherwise we will not be able to properly detect the platform we are running on (Marvell versus Intel DPU)

Reply:
@thom311 my thinking was this would be the single place these variables will be defined. I could define the constant for each subsequent DPU image in internal/platform/<dpu>.go if that makes more sense, and then add them here.

I want a single list of all support vsps somewhere so we can do: 

https://github.com/SalDaniele/dpu-operator/blob/deploy_vsp_from_daemon_v2/cmd/main.go#L101
https://github.com/SalDaniele/dpu-operator/blob/deploy_vsp_from_daemon_v2/cmd/daemon/daemon.go#L57

Reply:
Can we resolve the vsp image directly when we call `getVspImagesFromEnv` which we would rename to `vspImage() string` and put that into platform?